crates/sui-bridge-indexer/src/main.rs

```diff
@@ -73,64 +75,58 @@ async fn main() -> Result<()> {
     let bridge_metrics = Arc::new(BridgeMetrics::new(&registry));
 
     // unwrap safe: db_url must be set in `load_config` above
-    let db_url = config.db_url.clone().unwrap();
+    let db_url = config.db_url.clone();
 
     // TODO: retry_with_max_elapsed_time
     let eth_worker = EthBridgeWorker::new(
         get_connection_pool(db_url.clone()),
         bridge_metrics.clone(),
         indexer_meterics.clone(),
         config.clone(),
-    )
-    .unwrap();
+    )?;
 
     let eth_client = Arc::new(
         EthClient::<ethers::providers::Http>::new(
             &config.eth_rpc_url,
             HashSet::from_iter(vec![eth_worker.bridge_address()]),
             bridge_metrics.clone(),
         )
-        .await
-        .map_err(|e| anyhow::anyhow!(e.to_string()))?,
+        .await?,
     );
 
     let unfinalized_handle = eth_worker
         .start_indexing_unfinalized_events(eth_client.clone())
-        .await
-        .unwrap();
+        .await?;
     let finalized_handle = eth_worker
         .start_indexing_finalized_events(eth_client.clone())
-        .await
-        .unwrap();
+        .await?;
     let handles = vec![unfinalized_handle, finalized_handle];
 
     if let Some(sui_rpc_url) = config.sui_rpc_url.clone() {
-        start_processing_sui_checkpoints_by_querying_txes(
+        start_processing_sui_checkpoints_by_querying_txns(
             sui_rpc_url,
             db_url.clone(),
             indexer_meterics.clone(),
             bridge_metrics,
         )
-        .await
-        .unwrap();
+        .await?;
     } else {
-        let _ = start_processing_sui_checkpoints(
```

Comment:
I feel I'd leave this as is, which gives a bit more clarity?

Reply:
but this is not using the result which is wrong?

doing `let _ =` will simply ignore the result and we won't know if this task failed or not.